File: app/src/main/java/com/example/rustyalarm/alarm/AlarmEvent.kt
package com.example.rustyalarm.alarm

import androidx.room.Entity
import androidx.room.PrimaryKey

enum class AlarmEventType { FIRED, DISMISSED, SNOOZED }

@Entity(tableName = "alarm_events")
data class AlarmEvent(
    @PrimaryKey(autoGenerate = true)
    val id: Long = 0,
    val alarmId: Long,
    val eventType: String,                              // AlarmEventType.name
    val timestamp: Long = System.currentTimeMillis(),
    val challengeType: String = ChallengeType.NONE.name,
    val responseSeconds: Long? = null,                  // time between FIRED and DISMISSED
) {
    fun type(): AlarmEventType =
        runCatching { AlarmEventType.valueOf(eventType) }.getOrDefault(AlarmEventType.FIRED)
}

File: ios/Sources/Models/AlarmEvent.swift
import Foundation

enum AlarmEventType: String, Codable {
    case fired
    case dismissed
    case snoozed
}

struct AlarmEvent: Codable, Identifiable {
    let id: UUID
    let alarmId: UUID?
    let type: AlarmEventType
    let timestamp: Date
    let challengeType: ChallengeType?
    let responseSec: Int?

    init(
        id: UUID = UUID(),
        alarmId: UUID?,
        type: AlarmEventType,
        timestamp: Date = Date(),
        challengeType: ChallengeType? = nil,
        responseSec: Int? = nil,
    ) {
        self.id = id
        self.alarmId = alarmId
        self.type = type
        self.timestamp = timestamp
        self.challengeType = challengeType
        self.responseSec = responseSec
    }
}
